src/lib/components/Communities/CommunityHeader.js

```diff
@@ -0,0 +1,138 @@
+//
+// Invenio RDM Records is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React from 'react';
+import { Container, Image, Icon, Placeholder } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+
+class CommunityHeaderIcon extends React.Component {
+  render() {
+    return (
+      <Icon
+        className="ml-5"
+        link
+        color="grey"
+        onClick={() => console.log('TODO: Open community modal')}
+        name="setting"
+      />
+    );
+  }
+}
+
+class CommunityHeaderLoading extends React.Component {
+  render() {
+    return (
+      <Container className="community-header">
+        <Placeholder>
+          <Placeholder.Header image>
+            <Placeholder.Line />
+            <Placeholder.Line />
+          </Placeholder.Header>
+        </Placeholder>
+      </Container>
+    );
+  }
+}
+
+class CommunityHeaderInfo extends React.Component {
+  render() {
+    const { community, imagePlaceholderLink } = this.props;
+    const logoLink = community ? community?.links?.logo : imagePlaceholderLink;
+    return (
+      <Container className="community-header">
+        <Image size="mini" src={logoLink} /> {community?.metadata?.title}
+        <CommunityHeaderIcon />
+      </Container>
+    );
+  }
+}
+
+CommunityHeaderInfo.propTypes = {
+  imagePlaceholderLink: PropTypes.string.isRequired,
+  community: PropTypes.object,
+};
+
+CommunityHeaderInfo.defaultProps = {
+  community: undefined,
+};
+
+class EmptyCommunity extends React.Component {
+  render() {
+    const { error } = this.props;
+    return (
+      <Container className="community-header empty">
+        <span className={error && 'error-message'}>
+          {error
+            ? 'Failed to select community: ' + error.response.data.message
+            : 'No community selected.'}
+        </span>
+        <CommunityHeaderIcon />
+      </Container>
+    );
+  }
+}
+
+EmptyCommunity.propTypes = {
+  imagePlaceholderLink: PropTypes.string.isRequired,
+  community: PropTypes.object,
+  error: PropTypes.object,
+};
+
+EmptyCommunity.defaultProps = {
+  community: undefined,
+  error: undefined,
+};
+
+export class CommunityHeaderComponent extends React.Component {
+  componentDidMount() {
+    const { getCommunity, record, setCommunity } = this.props;
+    const searchOptions = window.location.search;
+    const communityId = new URLSearchParams(searchOptions).get('community');
+    // TODO: record.community is resolved? if not change it accordingly
+    if (record.community) {
+      setCommunity(record.community);
+    } else if (communityId) {
+      getCommunity(communityId);
+    }
+  }
+
+  render() {
+    const { loading, community, imagePlaceholderLink, error } = this.props;
+
+    return (
+      <Container className="deposits-community-header" fluid>
+        {loading ? (
+          <CommunityHeaderLoading />
```

Comment:
it would be more readable if we make `CommunityHeaderLoading` as HOC, see implementation in ILS of `Loader` component. It will help you to avoid at least one level of this conditional statement

Reply:
Hmm yes, I guess we can create this cmp in react-invenio-forms?, since it will be used widely (I guess)

Reply:
not needed anymore